### DISCUSSION TITLE
setting related data that is possibly undefined

### DISCUSSION THREAD

**Discussion by chrisdrackett** [prisma/prisma#2913] [Q&A] [answered]
we have the following code to handle updating a user. The `settingCalendarsToSync` value is optional. I'm curious if there is a better and/or other ways to write this as the speeding feels a little odd.

```ts
prisma.user.update({
          where: { id: userId },
          data: {
            name,
            ...(settingCalendarsToSync
              ? { settingCalendarsToSync: { set: settingCalendarsToSync } }
              : {}),
          },
        })
```

**Comment by ryands17** [comment, marked as answer]
Hey @chrisdrackett :wave: 
Unfortunately this is one of the options as you cannot directly pass `null` or `undefined` to that value.

There is also another way which you might want to handle like this:
```ts
 const val: Prisma.UserCreateInput['settingCalendarsToSync'] = {}
  settingCalendarsToSync && (val.set = settingCalendarsToSync)
  await prisma.user.create({
    data: {
      name: 'user1',
      settingCalendarsToSync: val,
    },
  })
```

But these can be the only workarounds at the moment.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
